document/js-api/index.bs

```diff
@@ -1356,20 +1358,19 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 </div>
 
 <div algorithm>
-The algorithm <dfn>IndexValueToU64</dfn>(|v|, |indextype|) asserts that a JavaScript value is the appropriate variant of {{IndexValue}} for an {{IndexType}}, and ensures that its value is in [=u64=] range for WebAssembly embedding operations, by performing the following steps:
+The algorithm <dfn>IndexValueToU64</dfn>(|v|, |indextype|) converts a JavaScript value to a WebAssembly [=u64=] for use in embedding operations. It is designed to act like [=[EnforceRange]=] [=unsigned long=] for {{IndexType}} "i32", and to extend these semantics to {{IndexType}} "i64", by performing the following steps:
 
 1. If |indextype| is "i32",
-    1. If |v| [=is a Number=],
-        1. Assert: Due to WebIDL types and [=[EnforceRange]=], 0 ≤ [=ℝ=](|v|) < 2<sup>64</sup>.
-        1. Return [=ℝ=](|v|) as a WebAssembly [=u64=].
-    1. Otherwise, [=throw=] a {{TypeError}}.
+    1. Let |n| be [=?=] [$ConvertToInt$](|v|, 32, "unsigned"), where the destination type is associated with [=[EnforceRange]=].
+
+        Note: This is equivalent to the [=js-unsigned-long|JS conversion rules=] for [=[EnforceRange]=] [=unsigned long=].
+    1. Return [=ℝ=](|n|) as a WebAssembly [=u64=].
 1. Else if |indextype| is "i64",
-    1. If |v| [=is a BigInt=],
-        1. If |v| is not equal to [=!=] [$ToBigUint64$](|v|), [=throw=] a {{TypeError}}.
+    1. Let |n| be [=?=] [$ToBigInt$](|v|).
+    1. If |n| is not equal to [=!=] [$ToBigUint64$](|n|), [=throw=] a {{TypeError}}.
```

Comment:
IIUC this is identical to saying `if n < 0 or n > 2^63-1, throw a TypeError`. Wouldn't that be simpler than referring to another helper?

Reply:
To me this feels like an elegant way of doing the range check, but I don't have a strong preference.

Reply:
I'd lean toward @backes's suggestion, but not strongly held either.